dts/bindings/adc/nxp,kinetis-adc16.yaml

```diff
@@ -20,8 +20,38 @@ properties:
     interrupts:
       required: true
 
+    dma-channel:
```

Comment:
Why isn't this specified via/in a `dmas` property?

Reply:
@galak , in edma the dma path is dmamux channel(SOC binded) -> dma channel(application defines)
according to dmas definition it can only have edma instance and edma mux channel,
'''
 dmas = <&edma0 0 40>
'''
so we need add dma-channel for each driver to use, specificlly to use PIT trigger the channel is limited to 0-4.

Reply:
Is there a reason not to add channel to the `dmas` specifier for edma?

Reply:
(for example the `dts/bindings/dma/nxp,lpc-dma.yaml` seems to define channel as part of the specifier for that dma engine)

Reply:
@galak the edma dts is is reused from previous definition in below
https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/tree/Bindings/dma/fsl-edma.txt 
so the dmas is as below
```
#- #dma-cells : Must be <2>.
#  The 1st cell specifies the DMAMUX(0 for DMAMUX0 and 1 for DMAMUX1).
#    Specific request source can only be multiplexed by specific channels
#    group called DMAMUX.
#  The 2nd cell specifies the request source(slot) ID.
#    See the SoC's reference manual for all the supported request sources.
dma-cells:
  - mux
  - source
```

Reply:
@hakehuang how does the linux driver than deal with the channel?  Since from what I can tell there is no similar property to what you are adding here.

Reply:
@galak , from the linux driver code the channel id is defined in structure 
see 
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/spi/spi-fsl-dspi.c?h=v5.11-rc5 
![image](https://user-images.githubusercontent.com/800198/105804966-f81c1a80-5fdb-11eb-9f08-45ff888e4fa6.png)
and used in 
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/dma/imx-dma.c?h=v5.11-rc5
![image](https://user-images.githubusercontent.com/800198/105805345-c2c3fc80-5fdc-11eb-9183-9d4eda3ba68b.png)

Reply:
So we either should have the dma driver allocate channel's or change the binding to have the channel encoded in the `dmas` property and make that difference between the Linux binding.

Reply:
add those in dmas is not a good option, because there is no reference to set the channel, as more drivers supports DMA there will be a potential conflicts,  so design a channel allocation is a better option, but this will have to change in the dma common code, Can I use another PR to fix it? and keep the code as is for this PR?

Reply:
I don't want to introduce a dts property that we know will go away.  So I'd suggest we focus on implementing the allocator scheme.  I think we can do that specific to the edma driver to start with and worry about making it more generic after.

Reply:
OK, Let me try to implement it in EDMA